src/preamble_sharedlib.js

```diff
@@ -4,5 +4,19 @@
 // Runtime essentials
 //========================================
 
+{{{
+(function() {
+  // add in RUNTIME_LINKED_LIBS, if provided
+  //
+  // for side module we only set Module.dynamicLibraries - and loading them
+  // will be handled by dynamic linker runtime in the main module.
+  if (RUNTIME_LINKED_LIBS.length > 0) {
+    return "if (!Module['dynamicLibraries']) Module['dynamicLibraries'] = [];\n" +
+           "Module['dynamicLibraries'] = " + JSON.stringify(RUNTIME_LINKED_LIBS) + ".concat(Module['dynamicLibraries']);\n";
+  }
+  return '';
```

Comment:
This closure syntax/technique looks strange to me but maybe this is just the way these preamble.js stuff is handled?   Is this s common pattern?

Reply:
Yes, it is kind of mimiced to similar place in `preamble.js`:

https://github.com/kripken/emscripten/blob/1.38.21-1-g46afd2308/src/preamble.js#L1610-L1619